add cuda benchmark ci

Gasoonjia opened the pull request:
Summary: Introduce cuda benchmark ci for monitoring cuda backend performance.

the ci will run on three situations:

1. it will run all possible models (voxtral, gemma and whisper) combine with all possible quantization schema on every day's 1am pst;
2. it will run an random model everytime a PR got merged;
3. manually tirggered by user.

Differential Revision: D87400561

larryliu0820 commented on .ci/scripts/cuda_benchmark.py:
```diff
@@ -0,0 +1,890 @@
+        "--fix_gpu_clock",
+        action="store_true",
+        help="Fix GPU clock frequency to maximum before benchmarking",
+    )
```
should we make this as default?

Gasoonjia commented on .github/workflows/cuda-perf.yml:
```diff
@@ -0,0 +1,393 @@
+        shell: bash
+        env:
+          # Default models for scheduled runs (all models) vs PR/manual runs
+          CRON_DEFAULT_MODELS: ${{ github.event_name == 'schedule' && 'mistralai/Voxtral-Mini-3B-2507,openai/whisper-small,openai/whisper-medium,openai/whisper-large-v3-turbo,google/gemma-3-4b-it' || 'openai/whisper-small' }}
```
Nice catch, now we statically use whisper-small for benchmarking jobs on PR. The latest commit have changed to randomly choose model every time.

Gasoonjia commented on .github/workflows/cuda-perf.yml:
```diff
@@ -0,0 +1,393 @@
+          echo "Generated benchmark configs:"
+          echo "$CONFIGS" | python -m json.tool
+
+  export-models:
```
i would keep it here for better control; we've reuse the same script.

Gasoonjia commented:
update dashborad content to make it more descriptive.

Gasoonjia committed:
add cuda benchmark ci (#15883)

Summary:

Introduce cuda benchmark ci for monitoring cuda backend performance.

the ci will run on three situations:

1. it will run all possible models (voxtral, gemma and whisper) on every day's 1am pst;

2. it will run an arbitrary model everytime a PR got merged;

3. manually tirggered by user.

Reviewed By: larryliu0820

Differential Revision: D87400561

Gasoonjia committed:
Merge branch 'main' into export-D87400561
